### DISCUSSION TITLE
请问如何安装浏览器插件

### DISCUSSION THREAD

**Discussion by shopconan** [ctripcorp/flybirds#32] [Q&A]
安装是一个
安装好了以后的点击呢

**Comment by clgwlg** [comment]
请问你指的浏览器插件，是PlayWright的录制插件codegen么？当前版本还未支持。
Flybirds的语法接近自然语言，上手成本还是蛮低的。
下面是一个例子
```Gherkin
功能: 乘机人模块

@p1 @android @web
场景:外露乘机人_选择列表页乘机人
   当   跳转页面到[单程填写页]
   那么 页面渲染完成出现元素[已选乘机人姓名]
   那么 [选择乘机人文案]的文案为[选择乘机人]
   那么 [已选乘机人姓名]的文案为[李易峰]
   那么 [已选乘机人证件类型]的文案为[护照]
   那么 [已选乘机人证件号]的文案为[YHE77]
   那么 存在[乘客类型标签儿童]的元素
   那么 返回上一页

```

**Reply by shopconan** [reply]
@clgwlg 
我现在有3种想法  不知道能不能实现 因为 我看到我们其实在airtest playwright上进行了封装，本意应该是简化流程

flybirds控制pc端的浏览器 逐次访问我提交的url，滚动 按照我预设的动作点击，录制整个操作过程
flybirds控制移动端的浏览器 逐次访问我提交的url，滚动 按照我预设的动作点击，录制整个操作过程
flybirds控制浏览器，加载浏览器插件，点击浏览器插件，比如similarweb之类，录制整个操作过程





功能: 网站首页

@p1 @android @web
场景:外露乘机人_选择列表页乘机人
   当   跳转页面到[网站首页]
  那么开始录屏
   那么  存在[cookie accept]的元素
   那么 [选择 cookie accept]的文案为[accept]
   那么  存在[coupon]的元素
   那么 [选择 coupon]的文案为[pass]
   那么 按照0.5的时间间隔，向下滚动直到页面底部
   那么结束录屏


功能: 网站首页

@p1 @android @web
场景:外露乘机人_选择列表页乘机人
   当   跳转页面到[网站首页]
  那么开始录屏
   那么  存在[cookie accept]的元素
   那么 [选择 cookie accept]的文案为[accept]
   那么  存在[coupon]的元素
   那么 [选择 coupon]的文案为[pass]
  那么点击选择[浏览器插件1]
 那么存在[浏览器插件的显示结果]
   那么 按照0.5的时间间隔，向下滚动直到页面底部
   那么结束录屏

**Comment by clgwlg** [comment]
Web端录制操作的例子，参考你上面的例子做了修改
```Gherkin
功能: 网站首页

@p1 @web
场景: 网站首页
   当   跳转页面到[网站首页]
   而且  页面渲染完成出现元素[cookie accept]             // 增加这一步，检查页面是否已渲染完成
   那么  选择 cookie accept]的文案为[accept]
   那么  存在[coupon]的元素
   那么  选择 coupon]的文案为[pass]
   那么  点击选择[浏览器插件1]
   那么  存在[浏览器插件的显示结果]
   那么  向[下] 查找[order]的元素      
   那么  结束录屏                 // web端只需在最后增加这句，就可以在报告中查看完整操作视频
```
关于滑动距离和时间的配置，参考[issue](https://github.com/ctripcorp/flybirds/issues/27)

**Comment by shopconan** [comment]
希望我尽快跑出来一个 哈哈

**Comment by shopconan** [comment]
@clgwlg 我前面提到的安卓/ios里跑浏览器 ，访问url这种可以吗
还是说建议直接在web端 类似firefox dev下调整窗口大小来模拟url在移动端的展示效果

**Reply by clgwlg** [reply]
建议直接在web端模拟移动端的展示效果。
PlayWright支持模拟移动端设备，可以增加一个DSL提供支持

**Comment by wanghaisheng** [comment]
@clgwlg  我如果没有记错 pl是没法操作地址栏水平线上的元素的，那我就没有办法点击设置，点击addon 搜索插件 安装

还是说这里可以用airtest的功能来实现？通过图像定位？

**Reply by clgwlg** [reply]
pl提供了方法，可以安装chrome extension
https://playwright.dev/python/docs/chrome-extensions

**Reply by wanghaisheng** [reply]
airtest 弄起来应该更灵活简单？

**Reply by clgwlg** [reply]
安装浏览器插件，pl提供的方法，我觉得比airtest的图像定位更稳定可靠
对浏览器插件的具体操作，可以考虑图像定位，或者直接修改插件的配置文件。

**Reply by wanghaisheng** [reply]
之前我遇到一个issue  就是地址栏左侧 https这里会出现一个get location的提醒 pl我死活就点击不了 提供的方法也无效。

**Comment by lc2lc2** [comment]
pl的点击方法，对这里是无效的，提供的extension安装方法试过么。

发自我的iPhone

在 2022年5月26日，13:38，HeisenBerg? ***@***.***> 写道：

﻿

之前我遇到一个issue 就是地址栏左侧 https这里会出现一个get location的提醒 pl我死活就点击不了 提供的方法也无效。

**Comment by wanghaisheng** [comment]
@lc2lc2 刚刚大神提的那个 我之前没有留意到过 还没试
之前卡在录屏清晰度有点久 本来我是用原生pl做，卡住了就四处找 看到flybirds 但一直也没有跑起来，昨晚算是跑出来一个demo 我也抛出来一些issue
